Scripts/Editor/Debug/AnvilDebugSafetyScriptDefines.cs

```diff
@@ -0,0 +1,75 @@
+
+        //For ANVIL_DEBUG_SAFETY, we don't require anything to turn it on, but we need to ensure that ANVIL_DEBUG_SAFETY_EXPENSIVE is turned off if we are off
+        private static readonly ScriptDefineDefinition ANVIL_DEBUG_SAFETY_DEFINITION = new ScriptDefineDefinition(ANVIL_DEBUG_SAFETY,
+                                                                                                                  ANVIL_DEBUG_SAFETY_MENU_PATH,
+                                                                                                                  null,
+                                                                                                                  new[]
+                                                                                                                  {
+                                                                                                                      ANVIL_DEBUG_SAFETY_EXPENSIVE
+                                                                                                                  });
+
+        //For ANVIL_DEBUG_SAFETY_EXPENSIVE, we require ANVIL_DEBUG_SAFETY to be on if we're on, but we don't need to turn anything off if we turn off
+        private static readonly ScriptDefineDefinition ANVIL_DEBUG_SAFETY_EXPENSIVE_DEFINITION = new ScriptDefineDefinition(ANVIL_DEBUG_SAFETY_EXPENSIVE,
+                                                                                                                            ANVIL_DEBUG_SAFETY_EXPENSIVE_MENU_PATH,
+                                                                                                                            new[]
+                                                                                                                            {
+                                                                                                                                ANVIL_DEBUG_SAFETY
+                                                                                                                            },
+                                                                                                                            null);
+
+        static AnvilDebugSafetyScriptDefines()
+        {
+            ScriptDefinesToggle.RegisterScriptDefineDefinition(ANVIL_DEBUG_SAFETY_DEFINITION);
+            ScriptDefinesToggle.RegisterScriptDefineDefinition(ANVIL_DEBUG_SAFETY_EXPENSIVE_DEFINITION);
+        }
+
+
+        [MenuItem(ANVIL_DEBUG_SAFETY_MENU_PATH, true)]
+        private static bool Toggle_Validator_Safety()
+        {
+            return ScriptDefinesToggle.Toggle_Validator(ANVIL_DEBUG_SAFETY_DEFINITION);
+        }
+
+        [MenuItem(ANVIL_DEBUG_SAFETY_MENU_PATH)]
+        private static void Toggle_Safety()
+        {
+            ScriptDefinesToggle.Toggle(ANVIL_DEBUG_SAFETY_DEFINITION);
+        }
```

Comment:
:bike: I've been putting the validator on the bottom. For consistency these should be but it's not a big deal.

Also, have been putting `_Validator` at the end so the method names more closely match.

Reply:
Ok sure, i'm not opposed to any of those. Done.